enhancements/ingress/empty-requests-policy.md

```diff
@@ -0,0 +1,267 @@
+	// detection and diagnosis of problems.  In addition, these requests may
+	// be caused by port scans, in which case logging empty requests may aid
+	// in detecting intrusion attempts.
+	//
+	// +optional
+	HTTPEmptyRequestsPolicy HTTPEmptyRequestsPolicy `json:"httpEmptyRequestsPolicy,omitempty"`
+}
+```
+
+The `HTTPEmptyRequestsPolicy` type accepts either one of two values: "Respond"
+and "Ignore":
+
+```go
+// HTTPEmptyRequestsPolicy indicates how HTTP connections for which no request
+// is received should be handled.
+// +kubebuilder:validation:Enum="";Respond;Ignore
+type HTTPEmptyRequestsPolicy string
+
+const (
+	// HTTPEmptyRequestsPolicyRespond indicates that the ingress controller
+	// should respond to empty requests.
+	HTTPEmptyRequestsPolicyRespond HTTPEmptyRequestsPolicy = "Respond"
+	// HTTPEmptyRequestsPolicyIgnore indicates that the ingress controller
+	// should ignore empty requests.
+	HTTPEmptyRequestsPolicyIgnore HTTPEmptyRequestsPolicy = "Ignore"
+)
+```
+
+By default, the IngressController does not log empty requests; empty requests
+are counted in metrics; and when an HTTP request times out without the
+IngressController's having received a response, the IngressController sends an
+HTTP 408 response and closes the connection.  The following example configures
+the IngressController not to log empty requests:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  logging:
+    access:
+      destination:
+        type: Container
+      logEmptyRequests: Ignore
+```
+
+The following example configures the IngressController to close HTTP connections
+without sending a response or counting them in metrics when they time out before
+a request is received:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  httpEmptyRequestsPolicy: Ignore
+```
+
+
+#### Validation
+
+Omitting either `spec.logging.access.logEmptyRequests` or
+`spec.httpEmptyRequestsPolicy` is valid and specifies the default behavior.  The
+API validates both fields as described by the respective field types'
+`+kubebuilder:validation:Enum` markers.
+
+### User Stories
+
+#### As a cluster administrator, I need to configure an IngressController not to log empty requests.
+
+To satisfy this use-case, the cluster administrator can set the
+IngressController's `spec.logging.access.logEmptyRequests` field to `Ignore`.
+
+#### As a cluster administrator, I need to configure an IngressController not to respond to empty HTTP requests.
+
+To satisfy this use-case, the cluster administrator can set the
+IngressController's `spec.httpEmptyRequestsPolicy` field to `Ignore`.
+
+### Implementation Details
+
+Implementing this enhancement requires changes in the following repositories:
+
+* openshift/api
+* openshift/cluster-ingress-operator
+* openshift/router
```

Comment:
Will the [router's metrics code ](https://github.com/openshift/router/blob/master/pkg/router/metrics/haproxy/haproxy.go)need to be modified to ignore empty requests when applicable? Or is there a drop in HAProxy setting that filters out empty requests from metrics automatically?

Reply:
The documentation for `option http-ignore-probes` states that the option does "prevent any error counter from being incremented".  The enhancement is really following the behavior of HAProxy in this respect.